hi-ml-azure/src/health_azure/utils.py

```diff
@@ -1316,6 +1331,33 @@ def is_completed(run_: Run) -> bool:
     return all(is_completed(run) for run in runs)
 
 
+def is_job_completed(job: Job) -> bool:
+    """Checks if the given AzureML v2 Job completed successfully.
+
+    :return: True if the job completed successfully, False for failures, job still running, etc."""
+    return job.status == JobStatus.Completed.value
+
+
+def wait_for_job_completion(ml_client: MLClient, job_name: str) -> None:
+    """Wait until the job of the given ID is completed or failed with an error. If the job did not complete
+    successfully, a ValueError is raised.
+
+    :param ml_client: An MLClient object for the workspace where the job lives.
+    :param job_name: The name (id) of the job to wait for.
+    :raises ValueError: If the job did not complete successfully (any status other than Completed)
+    """
+
+    while True:
+        # Get the latest job status by reading the whole job info again via the MLClient
+        updated_job = ml_client.jobs.get(name=job_name)
+        current_job_status = updated_job.status
+        if JobStatus.is_finished_state(current_job_status):
+            break
+        time.sleep(10)
```

Comment:
Why the sleep here?

Reply:
I think this is to stop the function bombarding the AzureML API with requests. 10s seems like a sensible time to wait between checking jobs statuses given that they can run for ~20-30 mins.